pyrobosim/pyrobosim/core/robot.py

```diff
@@ -207,11 +217,35 @@ def pick_object(self, obj_query):
             )
             return False
 
+        # If a grasp generator has been specified and no explicit grasp has been provided,
+        # generate grasps here.
+        # TODO: Specify allowed grasp types
+        if grasp_pose is not None:
+            self.latest_grasp = Grasp(origin=grasp_pose)
+        elif self.grasp_generator is not None:
+            cuboid_pose = Pose.from_transform(
+                np.matmul(obj.cuboid_pose.get_transform_matrix(),
+                          obj.pose.get_transform_matrix(),      
+                )
+            )
+            grasps = self.grasp_generator.generate(
+                obj.cuboid_dims, cuboid_pose, self.pose,
+                front_grasps=True, top_grasps=True, side_grasps=False
```

Comment:
Regarding the TODO above: How should we configure which grasps we allow for a robot?

One idea is to move this to the actual grasp generator object rather than to its `generate()` method, though that seems less flexible.

Reply:
I should also say, if you want to visualize the grasps while interacting with pyrobosim, you can add these lines of code below:

```python
            obj_footprint = np.array(list(obj.raw_polygon.exterior.coords))
            self.grasp_generator.show_grasps(
                obj.cuboid_dims, grasps, cuboid_pose,
                self.pose, obj_footprint)
```